docs/core-concepts/bot-sequences/actions.mdx

```diff
@@ -1,9 +1,659 @@
+replaying a sequence of static actions that were previously recorded as part of a play through.
+
+**Data Format**
+
+`type` is set to `InputPlayback`.
+
+```json
+// The `botAction.data` format
+{
+    "startTime": 0.0, // The time offset at which the action should be performed from the start of the segment
+    "inputData": {
+        "mouse": [ // Any number of mouse inputs can be provided
+            {
+                "startTime": ..., // A number indicating the time offset at which the action should be performed from the start of the segment (in seconds)
+                "screenSize": {"x": ..., "y": ...}, // A width and height indicating the size of the screen, which is used to linearly scale the position of the mouse when the screen size is different from the screen size at the time the inputs were recorded
+                "position": {"x": ..., "y": ...}, // A coordinate indicating the position to click, where X and Y are integers
+                "worldPosition": null, // An optional coordinate indicating the position to click in world space, which is often null.
+                "leftButton": ..., // A boolean indicating whether the left mouse button should be pressed
+                "middleButton": ..., // A boolean indicating whether the middle mouse button should be pressed
+                "rightButton": ..., // A boolean indicating whether the right mouse button should be pressed
+                "forwardButton": ..., // A boolean indicating whether the forward mouse button should be pressed
+                "backButton": ..., // A boolean indicating whether the back mouse button should be pressed
+                "scroll": {"x": ..., "y": ...}, // An optional scroll amount, where X and Y are integers
+                "clickedObjectNormalizedPaths": [...], // An optional list of normalized paths to the objects that were clicked. This is used to adjust the click coordinates when there are variances in resolution scaling or object position between runs.
+            }
+        ],
+        "keyboard": [ // Any number of keyboard inputs can be provided
+            {
+                "startTime": ..., // A number indicating the time offset at which the action should be performed from the start of the segment (in seconds)
+                "endTime": ..., // A number indicating the time offset at which the action should be stopped from the start of the segment (in seconds)
+                "binding": ..., // A sting representing the binding to be pressed
+            }
+        ]
+    }
+}
+```
+
+**Notes**
+
+- You can use the `startTime` and `endTime` fields to control the timing of actions. For instance, you can hold shift and press a key to result in a capital letter.
+- If two actions have the same `startTime`, the action that appears earlier in the list will be performed first.
+- You can omit an `endTime` of a keyboard action to have the action continue until it is disabled using using an action that only sets the `endTime`.
```

Comment:
```suggestion
- You can omit an `endTime` of a keyboard action to have that action continue until it is disabled by a future action which only sets the `endTime`.
```

Honestly still confusing to read, even after the suggestion. I don't want to spend too much time making it read well though unless we think it would be highly useful.